util/xs_scripts/parallel_sim.sh

```
@@ -224,7 +225,12 @@ function arg_wrapper() {
     dw=${args[4]}
     sample=${args[5]}
 
-    run $checkpoint $work_dir >$work_dir/$log_file 2>&1
+    if ! [[ "$GEM5_PERF_LOG_LIMIT_BYTES" =~ ^[0-9]+$ ]] || [ "$GEM5_PERF_LOG_LIMIT_BYTES" -lt 1 ]; then
+        echo "Invalid GEM5_PERF_LOG_LIMIT_BYTES='$GEM5_PERF_LOG_LIMIT_BYTES'" >&2
+        exit 1
+    fi
+
+    run $checkpoint $work_dir 2>&1 | tail -c "$GEM5_PERF_LOG_LIMIT_BYTES" >$work_dir/$log_file
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)